code/PostgreSQLQuery.php

```diff
@@ -58,8 +58,10 @@ public function __destruct()
 
     public function seek($row)
     {
-        pg_result_seek($this->handle, $row);
-        return $this->nextRecord();
+        // Specifying the zero-th record here will reset the pointer
+        $row = pg_fetch_array($this->handle, 0, PGSQL_NUM);
```

Comment:
You've overwritten `$row`?

Reply:
I don't understand why the `0` isn't `$row`.

Also I don't understand why this doesn't increment to the next (wrong) record next time nextRecord() is called.

The `nextRecord()` call after `seek()` must return the _same_ record in order to fix the bug in question.

Reply:
Yeah that was just a mistake 😖 . As I understand it (and from how it was working when testing it) if you call `pg_fetch_array` with a row number, you set the pointer to that location and get the record at that location. `pg_fetch_array` with a null row will return the row at the current pointer location and then increment the pointer.